### DISCUSSION TITLE
How to get the Hostnames of local hosts?

### DISCUSSION THREAD

**Discussion by rakor** [AdguardTeam/AdGuardHome#3753] [General]
Hey there,

my network is seperated VLANs using OpnSense as central firewall. So the OpnSense is dhcp-server for all subnets. Using unbound dns on the OpnSense to resolve only the local hostnames.

Currently I am running PiHole as central DNS. The IP of PiHole is propagated using the dhcp of the OpnSense.
The PiHole has the option "Use Conditional Forwarding" which I use to resolve local (.lan) domains using the OpnSense ans resolving all others from external dns.

How can I get the same result in AdGuardHome? Is there a way to set a "local dns server"?

Thanks

**Comment by ainar-g** [comment]
Hello.  If you mean reverse DNS, this is what the “Private reverse DNS servers” input on the “DNS settings” page is for.  Otherwise, if you mean the usual DNS queries, you can do that by adding a custom upstream for your domain.  Something like:

```none
[/lan/]192.168.12.34
```

**Reply by rakor** [reply]
Thanks for your reply. No I don't think about reverse DNS. Reverse DNS, as far as I know, gives me the domainname for a given IP. But I want the IP for a given hostname, so I can say `ping somehost.lan` inside my network. So DNS but for my local hosted machines.
